[Deps] Update `automattic-tracks` to 6.0.4

ParaskP7 opened the pull request:
Closes: [AINFRA-918](https://linear.app/a8c/issue/AINFRA-918/snandroid)

---

### Description
<!--
***(Required)*** Add a concise description of what you fixed.  If this is related to an issue, add a link to it.  If applicable, add screenshots, animations, or videos to help illustrate the fix.
-->

This PR updates `automattic-tracks` to [6.0.4](https://github.com/Automattic/Automattic-Tracks-Android/releases/tag/6.0.4).

FYI: This update makes this library fully 16 KB page size compatible ([context](https://android-developers.googleblog.com/2025/05/prepare-play-apps-for-devices-with-16kb-page-size.html)).

PS: Also see:

- Kotlin [1.9.25](https://github.com/JetBrains/kotlin/releases/tag/v1.9.25): f14db5acf4b679a0de4addf22b894b7870bb70dd
- Dagger [2.52](https://github.com/google/dagger/releases/tag/dagger-2.52): dbfb1a464401ece197df1e79b2c4e8be87b016fe
- AndroidX Lifecycle [2.5.1](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.5.1): 7b61292c3f98255fdec3ddc48dd4f02c3db83ba7
- Error Resolution: 6a364d280a08f3f6eeb4542c662ce7b390ca9eb0 + 27992a68bd078e7f1a397e0af027a2a9ce9d85ec + 57e3f58f5472959de3324f85159fbe7fa3eb0446 + 22ef40710a909499ca71f819b44ec3b81f2e17c0
- Theme Fix: 41ea3e10bdada04b31861707327d4074bbe22afb
- Coroutine Tests: c477a63e6e13cc81213f461e7c78e65de71ecdd2 + 42d907a4de1180880c6cb9a101c3dfc38a1c5251

---

### Test
<!--
***(Required)*** List the steps to test the behavior.  For example:
1. Go to...
2. Tap on...
3. See error...
-->

#### Tracks & Events

1. Using the patch below, add tracking logs to the app.

<details>
    <summary>patch</summary>

```diff
Index: Simplenote/src/main/java/com/automattic/simplenote/analytics/AnalyticsTrackerNosara.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Simplenote/src/main/java/com/automattic/simplenote/analytics/AnalyticsTrackerNosara.java b/Simplenote/src/main/java/com/automattic/simplenote/analytics/AnalyticsTrackerNosara.java
--- a/Simplenote/src/main/java/com/automattic/simplenote/analytics/AnalyticsTrackerNosara.java	(revision 41ea3e10bdada04b31861707327d4074bbe22afb)
+++ b/Simplenote/src/main/java/com/automattic/simplenote/analytics/AnalyticsTrackerNosara.java	(date 1753449557344)
@@ -3,6 +3,7 @@
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.text.TextUtils;
+import android.util.Log;
 
 import androidx.preference.PreferenceManager;
 
@@ -95,8 +96,11 @@
         if (properties != null) {
             JSONObject propertiesJson = new JSONObject(properties);
             mNosaraClient.track(EVENTS_PREFIX + eventName, propertiesJson, user, userType);
+            String jsonString = propertiesJson.toString();
+            Log.i("AnalyticsTrackerNosara", "\uD83D\uDD35 Tracked: " + eventName + ", Properties: " + jsonString);
         } else {
             mNosaraClient.track(EVENTS_PREFIX + eventName, user, userType);
+            Log.i("AnalyticsTrackerNosara", "\uD83D\uDD35 Tracked: " + eventName);
         }
     }
 
@@ -128,4 +132,4 @@
 
         mNosaraClient.flush();
     }
-}
\ No newline at end of file
+}
```
</details>

2. Navigate to `Setting` -> `Privacy` section and make sure to enable `Share analytics`.
3. Open `Logcat` and filter for `🔵 Tracked: ` to make sure that all specific app actions are tracked (ie. switching tabs).

#### Sentry & Crashes

1. Navigate to `Setting` -> `Privacy` section and make sure to enable `Share analytics`.
2. Using the patch below, make the app crash. You could navigate to `Settings` -> `About` section and then click `Crash The App`.

<details>
    <summary>patch</summary>

```diff
Subject: [PATCH] Fix: Add an default simplestyle splash them point to simplestyle  

FYI: This change is done because after all the updates, the app started 
showing a blue overlay with an icon on every screen. With this fix and 
for Android 12 devices and above (API 31) we are no longer dependending 
on this custom splash screen theme, but rather let the system default to
whichever is the default implementation of splashscreen, without us 
adding any additional configuration on top of it and thus defaulting the
main application theme to 'Theme.Simplestyle' for Android 12 devices and
above.
---
Index: Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java b/Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java
--- a/Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java	(revision 41ea3e10bdada04b31861707327d4074bbe22afb)
+++ b/Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java	(date 1753449660581)
@@ -169,6 +169,17 @@
             }
         });
 
+        findPreference("pref_key_crash_the_app").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+            @Override
+            public boolean onPreferenceClick(Preference preference) {
+                try {
+                    throw new Exception("Developer caused a crash!");
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
         findPreference("pref_key_about").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
             @Override
             public boolean onPreferenceClick(Preference preference) {
Index: Simplenote/src/main/res/xml/preferences.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Simplenote/src/main/res/xml/preferences.xml b/Simplenote/src/main/res/xml/preferences.xml
--- a/Simplenote/src/main/res/xml/preferences.xml	(revision 41ea3e10bdada04b31861707327d4074bbe22afb)
+++ b/Simplenote/src/main/res/xml/preferences.xml	(date 1753449660581)
@@ -190,6 +190,12 @@
             android:title="@string/website">
         </Preference>
 
+        <Preference
+            android:key="pref_key_crash_the_app"
+            android:persistent="false"
+            android:title="Crash the App">
+        </Preference>
+
     </PreferenceCategory>
 
 </androidx.preference.PreferenceScreen>
```

</details>

3. Open [Sentry](https://a8c.sentry.io) and navigate to `Issues`. Find `simplenote-android` and check the latest `debug` exceptions. You could search for `Developer caused a crash!` and check the latest crash, making sure that the crash is reported.

#### Smoke Test

Just because this update also includes multiple other core updates (`Kotlin`, `Dagger` and `AndroidX Lifecycle`), please smoke test the app fully and verify everything is working as expected.

---

### Review
<!--
***(Required)*** Add instructions for reviewers.  For example:
Only one developer and one designer are required to review these changes, but anyone can perform the review.
-->

FYI: Updating [automattic-tracks to 6.0.4](https://github.com/Automattic/simplenote-android/pull/1731/commits/e7afdf5949ae8a6cf413c980e50704ebe0b8e666) wasn't enough by itself to make Sentry fully 16 KB page size compatible. Updating [sentry plugin to 5.8.0](https://github.com/Automattic/simplenote-android/pull/1731/commits/c9159f17f81f20d595b49276ba212fa2d202866b) as well did the trick.

PS: I noticed this while working on [AINFRA-994](https://linear.app/a8c/issue/AINFRA-994), after I manually updated `Gradle to 8.13.4` and `AGP to 8.11.1`, then run `lintDebug`, only to notice Sentry still in the [Aligned16KB](https://googlesamples.github.io/android-custom-lint-rules/checks/Aligned16KB.md.html) list.

---

### Release
<!--
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
`RELEASE-NOTES.txt` was updated in d3adb3ef with:
> Added markdown support
-->
<!--
If the changes should not be included in release notes, add a statement to this section. For example:
These changes do not require release notes.
-->

`TODO` (I am actually not sure what to do here, maybe this instruction is now legacy)

ParaskP7 committed:
Deps: Update automattic-tracks to 6.0.4

Release Notes: https://github.com/Automattic/Automattic-Tracks-Android/
releases/tag/6.0.4

FYI: This update makes this library fully 16 KB page size compatible.

ParaskP7 committed:
Sentry: Delegate creating a release name to track library and sentry

PS: Instead of providing value from our build config.

ParaskP7 committed:
Sentry: Disable crash logging logs by default to reduce sentry noise

ParaskP7 committed:
Deps: Update kotlin to 1.9.25

Release Notes: https://github.com/JetBrains/kotlin/releases/tag/v1.9.25

FYI: This is done because 'automattic-tracks' itself is on Kotlin
version '1.9.24', which makes the current Kotlin version for this
project incompatible, see build failure below:

"Class 'kotlin.Unit' was compiled with an incompatible version of
Kotlin. The binary version of its metadata is 1.9.0, expected version
is 1.6.0."

PS: I chose '1.9.25' instead of '1.9.24' because that version is the
latest Kotlin '1.9' version before Kotlin '2.0'.

ParaskP7 committed:
Deps: Update dagger to 2.52

Release Notes: https://github.com/google/dagger/releases/tag/dagger-2.52

FYI: This is done because of the Kotlin '1.9.25' update, which makes it
incompatible with the current '2.38.1' and/or '2.40.1' version of
Dagger, see build failure below:

"Unsupported metadata version. Check that your Kotlin version is
>= 1.0: java.lang.IllegalStateException: Unsupported metadata version.
Check that your Kotlin version is >= 1.0"

PS: I chose '2.52' instead of the latest '2.57' because that version
is the latest Dagger version compatible with Kotlin '1.9'.

ParaskP7 committed:
Deps: Update androidx lifecycle to 2.5.1

Release Notes: https://developer.android.com/jetpack/androidx/releases/
lifecycle#2.5.1

FYI: This is done because of the Dagger '2.52' update, which makes it
incompatible with the current '2.3.1' version of AndroidX Lifecycle, see
build failure below:

"Duplicate class androidx.lifecycle.ViewModelLazy/
androidx.lifecycle.ViewTreeViewModelKt found in modules 2.3.1 and 2.5.1"

PS: I chose '2.5.1' instead of the latest '2.9.2' because that version
is compatible with Kotlin '1.9', but also because I didn't want to
introduce more (unnecessary) changes to this project. For example, I
could have chosen '2.8.7' as well, which also is fully compatible with
Kotlin '1.9', but I could have chosen the latest '2.9.2' because this
version is only compatible with Kotlin '2+'.

ParaskP7 committed:
Analysis: Resolve exhaustive when expression related errors

Error Messages: "'when' expression must be exhaustive, add necessary
'null' branch or 'else' branch instead"

ParaskP7 committed:
Analysis: Resolve moved to val related errors

Error Messages:
- "Using 'body(): ResponseBody?' is an error. moved to val"
- "Using 'code(): Int' is an error. moved to val"

ParaskP7 committed:
Analysis: Resolve media type related error

Error Message: "MediaType?' is an error. moved to extension function"

ParaskP7 committed:
Analysis: Resolve offer versus try send related error

Error Messages:
- "Using 'offer(E): Boolean' is an error. Deprecated in the favour of
'trySend' method"
- "Using 'offer(E): Boolean' is an error. Deprecated in the favour of
'trySend' method"

ParaskP7 committed:
Fix: Add an default simplestyle splash them point to simplestyle

FYI: This change is done because after all the updates, the app started
showing a blue overlay with an icon on every screen. With this fix and
for Android 12 devices and above (API 31) we are no longer depending on
this custom splash screen theme, but rather let the system default to
whichever is the default implementation of splashscreen, without us
adding any additional configuration on top of it and thus defaulting the
main application theme to 'Theme.Simplestyle' for Android 12 devices and
above.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
Test: Fix coroutine related failing tests

FYI: This change also introduces 'CoroutineTestRule' to this project.

ParaskP7 committed:
Test: Replace deprecated test coroutine dispatcher and run blocking test

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/utils/crashlogging/SimplenoteCrashLoggingDataProvider.kt:
```diff
@@ -4,33 +4,42 @@ import com.automattic.android.tracks.crashlogging.CrashLoggingDataProvider
 
     override val buildType = BuildConfig.BUILD_TYPE
-    override val enableCrashLoggingLogs = BuildConfig.DEBUG
+    override val enableCrashLoggingLogs = false
```
This is done by design: dcbac287b3d745c84c4933bdd0a616ac64c9d2e7

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/utils/crashlogging/SimplenoteCrashLoggingDataProvider.kt:
```diff
@@ -4,33 +4,42 @@ import com.automattic.android.tracks.crashlogging.CrashLoggingDataProvider
-    override fun applicationContextProvider(): Map<String, String> {
-        return emptyMap()
-    }
+    override val user: Flow<CrashLoggingUser?> = MutableStateFlow(app.simperium?.user?.toCrashLoggingUser())
```
Yea, now that you mention it, I am actually not too sure about this, hmmm... 🤔 Cc @wzieba

wzieba replied:
the AI suggestion is correct

ParaskP7 replied:
Done: 73129c3853c1fff23723e22acbc710341e9aeb5d @wzieba

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumTagsRepository.kt:
```diff
@@ -82,9 +81,9 @@ class SimperiumTagsRepository @Inject constructor(
-        val callbackOnSaveObject = Bucket.OnSaveObjectListener<Tag> { _, _ -> offer(true) }
-        val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Tag> { _, _ -> offer(true) }
-        val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Tag> { _, _, _ -> offer(true) }
+        val callbackOnSaveObject = Bucket.OnSaveObjectListener<Tag> { _, _ -> trySend(true).isSuccess }
+        val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Tag> { _, _ -> trySend(true).isSuccess }
+        val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Tag> { _, _, _ -> trySend(true).isSuccess }
```
I think we are okay with that, thanks though! 👍

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumTagsRepository.kt:
```diff
@@ -82,9 +81,9 @@ class SimperiumTagsRepository @Inject constructor(
-        val callbackOnSaveObject = Bucket.OnSaveObjectListener<Tag> { _, _ -> offer(true) }
-        val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Tag> { _, _ -> offer(true) }
-        val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Tag> { _, _, _ -> offer(true) }
+        val callbackOnSaveObject = Bucket.OnSaveObjectListener<Tag> { _, _ -> trySend(true).isSuccess }
+        val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Tag> { _, _ -> trySend(true).isSuccess }
+        val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Tag> { _, _, _ -> trySend(true).isSuccess }
```
I think we are okay with that, thanks though! 👍

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumTagsRepository.kt:
```diff
@@ -82,9 +81,9 @@ class SimperiumTagsRepository @Inject constructor(
-        val callbackOnSaveObject = Bucket.OnSaveObjectListener<Tag> { _, _ -> offer(true) }
-        val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Tag> { _, _ -> offer(true) }
-        val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Tag> { _, _, _ -> offer(true) }
+        val callbackOnSaveObject = Bucket.OnSaveObjectListener<Tag> { _, _ -> trySend(true).isSuccess }
+        val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Tag> { _, _ -> trySend(true).isSuccess }
+        val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Tag> { _, _, _ -> trySend(true).isSuccess }
```
I think we are okay with that, thanks though! 👍

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumCollaboratorsRepository.kt:
```diff
@@ -69,17 +69,17 @@ class SimperiumCollaboratorsRepository @Inject constructor(
-                offer(true)
+                trySend(true).isSuccess
             }
         }
         val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Note> { _, note ->
             if (note.simperiumKey == noteId) {
-                offer(true)
+                trySend(true).isSuccess
             }
         }
         val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Note> { _, _, updatedNoteId ->
             if (updatedNoteId != null && noteId == updatedNoteId) {
-                offer(true)
+                trySend(true).isSuccess
```
I think we are okay with that, thanks though! 👍

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumCollaboratorsRepository.kt:
```diff
@@ -69,17 +69,17 @@ class SimperiumCollaboratorsRepository @Inject constructor(
-                offer(true)
+                trySend(true).isSuccess
             }
         }
         val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Note> { _, note ->
             if (note.simperiumKey == noteId) {
-                offer(true)
+                trySend(true).isSuccess
             }
         }
         val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Note> { _, _, updatedNoteId ->
             if (updatedNoteId != null && noteId == updatedNoteId) {
-                offer(true)
+                trySend(true).isSuccess
```
I think we are okay with that, thanks though! 👍

ParaskP7 commented on Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumCollaboratorsRepository.kt:
```diff
@@ -69,17 +69,17 @@ class SimperiumCollaboratorsRepository @Inject constructor(
-                offer(true)
+                trySend(true).isSuccess
             }
         }
         val callbackOnDeleteObject = Bucket.OnDeleteObjectListener<Note> { _, note ->
             if (note.simperiumKey == noteId) {
-                offer(true)
+                trySend(true).isSuccess
             }
         }
         val callbackOnNetworkChange = Bucket.OnNetworkChangeListener<Note> { _, _, updatedNoteId ->
             if (updatedNoteId != null && noteId == updatedNoteId) {
-                offer(true)
+                trySend(true).isSuccess
```
I think we are okay with that, thanks though! 👍

ParaskP7 committed:
Deps: Update sentry plugin to 5.8.0

Release Notes: https://github.com/getsentry/
sentry-android-gradle-plugin/releases/tag/5.8.0

FYI: This update makes this library fully 16 KB page size compatible.

------------------------------------------------------------------------

As part of this e7afdf5949ae8a6cf413c980e50704ebe0b8e666 change, I
forgot to also update the 'Sentry' plugin, which is actually required to
make Sentry fully 16 KB page size compatible. I noticed that when I
manually updated Gradle to 8.14.3 and AGP 8.11.1 to double check the
'Aligned16KB' rule, which still showed 'Sentry' on the list of
non-compatible libraries (AINFRA-994).

ParaskP7 committed:
Deps: Update screengrab to 2.1.1

Release Notes: Not published as a standalone document

FYI: Otherwise, running 'lintDebug' fails with the below build failure
on the 'com.jraska:falcon' transitive dependency:

For more info see:
- [screengrab] version bump lib to 2.1.1 #19483
https://github.com/fastlane/fastlane/pull/19483
- Publish screengrab without dependencies to jcenter repository #19480
https://github.com/fastlane/fastlane/issues/19480

------------------------------------------------------------------------

> Task :Simplenote:generateDebugLintReportModel FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':Simplenote:generateDebugLintReportModel'.
> Could not resolve all files for configuration
 ':Simplenote:debugAndroidTestCompileClasspath'.
   > Could not find com.jraska:falcon:2.1.1.
     Required by:
         project :Simplenote > tools.fastlane:screengrab:2.0.0

ParaskP7 committed:
Fix: Always provide an updated user for crash logging purposes

PR Comment: https://github.com/Automattic/simplenote-android/pull/1731
#discussion_r2231064987

mzorz commented:
hihi! after all the dance to make it build locally (thank you @ParaskP7  for all he help!!), building the branch and installing the app from scratch on a Google Pixel 8 with Android 14 builds but then tapping on the login button produces this crash:

```
2025-07-30 19:31:32.910 26302-26302 AndroidRuntime          com.automattic.simplenote.debug      E  FATAL EXCEPTION: main (Ask Gemini)
                                                                                                    Process: com.automattic.simplenote.debug, PID: 26302
                                                                                                    java.lang.IllegalArgumentException: CreationExtras must have a value by `SAVED_STATE_REGISTRY_OWNER_KEY`
                                                                                                    	at androidx.lifecycle.SavedStateHandleSupport.createSavedStateHandle(SavedStateHandleSupport.kt:91)
                                                                                                    	at dagger.hilt.android.internal.lifecycle.HiltViewModelFactory$2.create(HiltViewModelFactory.java:100)
                                                                                                    	at dagger.hilt.android.internal.lifecycle.HiltViewModelFactory.create(HiltViewModelFactory.java:170)
                                                                                                    	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:187)
                                                                                                    	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:153)
                                                                                                    	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:53)
                                                                                                    	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:35)
                                                                                                    	at com.automattic.simplenote.authentication.SignInFragment.getViewModel(SignInFragment.kt:41)
                                                                                                    	at com.automattic.simplenote.authentication.SignInFragment.onCreateView(SignInFragment.kt:127)
                                                                                                    	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:3104)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:524)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
                                                                                                    	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1890)
                                                                                                    	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1808)
                                                                                                    	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1751)
                                                                                                    	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2976)
                                                                                                    	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2886)
                                                                                                    	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:263)
                                                                                                    	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:351)
                                                                                                    	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
                                                                                                    	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1701)
                                                                                                    	at android.app.Activity.performStart(Activity.java:9003)
                                                                                                    	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:4018)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:327)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:307)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:279)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:150)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:93)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2595)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:107)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8592)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
```

Apparently this is due to a mismatch in some of the navigation libraries - I'm investigating on the side, will try and put up a PR on top of this one to fix it.

Aside from that I observed the changes and these LGTM 👍

ParaskP7 commented:
👋 @mzorz and thanks for reviewing/testing these changes! 🙇

> hihi! after all the dance to make it build locally (thank you @ParaskP7 for all he help!!), building the branch and installing the app from scratch on a Google Pixel 8 with Android 14 builds but then tapping on the login button produces this crash:

Oh, this is a great find, I just reproduced it and it crashes on my side as well, the very same crash! 🎯

I didn't think to test the logged-out flow, many thanks for catching that! 🙏

> Apparently this is due to a mismatch in some of the navigation libraries - I'm investigating on the side, will try and put up a PR on top of this one to fix it.

Ah thanks so much for doing so! 🙇

If you haven't gone too far with this, and want to, I can take a look instead, try to fix this, just let me know. Again, thanks for the help! 🙏

PS: Honestly, I think it is such a pain to make such changes to such an unmaintained library, the dependency hell is so obvious in here. Thus, I would again suggest we add this process we discussed for this project as soon as possible: p1753454031304299/1753453412.343859-slack-C027TG8D1BL

> Aside from that I observed the changes and these LGTM 👍

👍

mzorz commented:
> PS: Honestly, I think it is such a pain to make such changes to such an unmaintained library, the dependency hell is so obvious in here. Thus, I would again suggest we add this process we discussed for this project as soon as possible: p1753454031304299/1753453412.343859-slack-C027TG8D1BL

Definitely, my upcoming PR will try to update everything and smoke test, so we can take it from there 👍

mzorz reviewed (approved):
I'm approving the PR @ParaskP7  - the crashes are fixed on the upcoming PR #1733, so maybe we want to merge that one first, and then this one can land on `trunk` 👍

ParaskP7 commented:
👋 @mzorz !

> I'm approving the PR @ParaskP7 - the crashes are fixed on the upcoming PR https://github.com/Automattic/simplenote-android/pull/1733, so maybe we want to merge that one first, and then this one can land on trunk 👍

Awesome, great, I agree and thanks, yes, let's merge this, and then deal with #1733 to fix the crash. There isn't any scheduled release for this "in-maintenance" app, so there is not much risk in doing so. 👍 🚀

ParaskP7 commented:
Just a friendly nudge on this @wzieba in case you also want to quick check this, no matter if it's now merged! 😊 🙏
